### PR TITLE
I factorized the module that converts tracker/calo fragments into digit

### PR DESCRIPTION
**Pull request by gianipez**
I left around also the old module to make life simpler for some validation tests necessary for other developments that will come soon by other users. One of them is Stefano Miscetti who has been working with me and Ray in developing the ProditionsService for the calorimeter channel-map. I plan to remove the old module once that development is done.

### CODE REVIEW

**Review by kutschke — COMMENTED** [review]
As I read this you split one module into two.  So now we have two copies of lots of code.  Is there a plan to delete the old code once the new ones are tested? If not, please create classes to do the duplicated work and use those classes inside the modules.  In this pattern the modules are thing wrappers around algorithm classes.

**Comment by kutschke** [comment]
OOps.  I see that your initial comments already say that you plan to remove the old one.  So never mind that comment.

File: DAQ/src/CaloRecoFromFragments_module.cc [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  // Store the calo digis in the event
+  event.put(std::move(calo_digis));
+
+
+} // produce()
+
+void art::CaloRecoFromFragments::analyze_calorimeter_(
+    const artdaq::Fragment& f, std::unique_ptr<mu2e::CaloDigiCollection> const& calo_digis) {
+
+  mu2e::CalorimeterFragment cc(f);
```
[the remark this answers is not in the document]
Comment (gianipez) [reply DAQ/src/CaloRecoFromFragments_module.cc:118]: it is not. It's just wrapping the Fragment:
https://cdcvs.fnal.gov/redmine/projects/mu2e_artdaq/repository/core/revisions/develop/entry/mu2e-artdaq-core/Overlays/ArtFragment.hh
https://cdcvs.fnal.gov/redmine/projects/mu2e_artdaq/repository/core/revisions/develop/entry/mu2e-artdaq-core/Overlays/CalorimeterFragment.hh

File: DAQ/src/CaloRecoFromFragments_module.cc [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+  if (diagLevel_ > 1) {
```
[the remark this answers is not in the document]
Comment (gianipez) [reply DAQ/src/CaloRecoFromFragments_module.cc:120]: Thanks for the suggestion! Can you point me to an example?

File: DAQ/src/CaloRecoFromFragments_module.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (kutschke) [reply DAQ/src/CaloRecoFromFragments_module.cc:118]: Thanks.  I confused artdaq::Fragment and ArtFragment.  Now it's clear.

File: DAQ/src/CaloRecoFromFragments_module.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (kutschke) [reply DAQ/src/CaloRecoFromFragments_module.cc:120]: This is the way I would do it.  Add a new buildopt to inject -DNOTRIGGERDIAG into the compiler options.   Then wrap the code blocks in:

<pre>
#ifndef NOTRIGGERDIAG  
  // code goes here
#endif
</pre>

Note that the idiom is that the default is on and to turn it off you set NOTRIGGERDIAG.  The price of this is the double negative in the code.   The alternative you could make -DTRIGGERDIAG the default and provide a buildopts switch to remove it.

**Comment by kutschke** [comment]
Adding a comment to reset the status of the build flag.

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Pull request merged**